### ISSUE TITLE
Unify JsonStream and document APIs

### ISSUE THREAD

**Issue by jkeiser** [lemire/simdjson#516]
JsonStream needs to be at least renamed for capitalization consistency. But there's an opportunity here to make the interface similar to the existing one in terms of error handling and iteration. Here's what I think:

```c++
document::parser parser;
parser.allocate_capacity(1024); // All individual documents are < 1K
auto [documents, error] = parser.parse_many(get_corpus("giant_file.json"));
for (auto [document, error] : documents) {
  if (error) { exit(1); }
  // use document
}
```

**Comment by lemire** [comment]
Agreed. That would be a great API but we have to test out for performance. It might be necessary to do something like...

```C++
for (auto& [document, error] : documents) {
  if (error) { exit(1); }
  // use document
}
```
... or something equivalent so that we don't waste time copying stuff that should not get copied.

**Comment by jkeiser** [comment]
To be clear, the doc there would be const document &, just like with the parser.parse API. So I think we're covered on the performance front there.

**Comment by lemire** [comment]
Good.

**Comment by jkeiser** [comment]
Fixed by #534.
